src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -97,19 +98,78 @@ ScheduleMap GcsPackStrategy::Schedule(
 ScheduleMap GcsSpreadStrategy::Schedule(
     std::vector<std::shared_ptr<ray::BundleSpecification>> &bundles,
     const std::unique_ptr<ScheduleContext> &context) {
+  // When selecting nodes, if you traverse from the beginning each time, a large number of
+  // bundles will be deployed to the previous nodes. So we start with the last selected
+  // node.
   ScheduleMap schedule_map;
-  auto &alive_nodes = context->node_manager_.GetClusterRealtimeResources();
-  auto iter = alive_nodes.begin();
-  size_t index = 0;
-  size_t alive_nodes_size = alive_nodes.size();
-  for (; iter != alive_nodes.end(); iter++, index++) {
-    for (size_t base = 0;; base++) {
-      if (index + base * alive_nodes_size >= bundles.size()) {
+  auto node_resources = context->node_manager_.GetClusterRealtimeResources();
+  if (node_resources.empty()) {
+    return schedule_map;
+  }
+
+  auto candidate_nodes = node_resources;
+  auto iter = candidate_nodes.begin();
+  auto iter_begin = iter;
```

Comment:
Why don't we just store begin and end in the beginning of the function and do iteration 
begin -> end
end -> begin? 

It is the same logic as your current one right?

Reply:
begin needs to be updated in the for loop.The current logic is to get the node that meet the resource requirements after the last selected node.